tests/unit/struct_inheritance.hlsl.expected

```diff
@@ -0,0 +1,10 @@
+template<bool Value>
+struct bool_literal{static const bool value=Value;};
+struct true_type:bool_literal<true>{};
+struct false_type:bool_literal<false>{};
+struct is_floating_point:false_type{};
+template<>
+struct is_floating_point<float>:true_type{};
```

Comment:
@eldritchconundrum should the template defn be emitted on the same line? does that matter for minification?

Reply:
It matters a little because it affects the compression of the generated shader.
In the printer, we use \n when the newline is mandatory (the shader won't work without it), and we use \0 to mean "unessential" newline that can be removed by the formatter when no indentation is needed (e.g. with `--format text`).

Reply:
ah got it, i guess that would require not having the template definition in a verbatim block though